code/modules/modular_computers/hardware/printer.dm

```diff
@@ -50,6 +50,42 @@
 		qdel(I)
 		stored_paper++
 		return TRUE
+	if(istype(I, /obj/item/paper_bin))
+		var/obj/item/paper_bin/bin = I
+		if(LAZYLEN(bin.papers))
+			if(stored_paper >= max_paper)
+				to_chat(user, span_warning("You try to feed \the [bin] into [src]'s paper recycler, but its paper bin is full!"))
+				return FALSE
+			/// Number of sheets we're adding
+			var/num_to_add = 0
+			/// Some goober put their manifesto in the paperbin, complain at the user if that's why nothing happened
+			var/rejected_sheets = FALSE
+			for(var/obj/item/paper/the_paper in bin.papers) // Search for the first blank sheet of paper, then toss it in
+				if(the_paper.info != "") // Uh oh, paper has words! 
+					rejected_sheets = TRUE
+					continue
+				if(istype(the_paper, /obj/item/paper/carbon)) // Add both the carbon, and the copy
+					var/obj/item/paper/carbon/carbon_paper = the_paper
+					if(!carbon_paper.copied && ((max_paper - stored_paper) >= 2)) // See if there's room for both
+						num_to_add = 2
+				else
+					num_to_add = 1
+				LAZYREMOVE(bin.papers, the_paper)
+				qdel(the_paper)
+				stored_paper += num_to_add
+				break // All full!
+			bin.update_appearance()
+			if(!num_to_add)
+				if(rejected_sheets)
+					to_chat(user, span_warning("The [src]'s paper recycler detects writing on everything in \the [bin], outright refusing to accept anything!"))
```

Comment:
I was going to change this to a balloon alert, but now I'm just confused. Writing on *everything* in the bin?

Reply:
There's always the possibility someone emptied out the bin and stuffed it full of stuff with writing on it. It'll proc whether its just one sheet someone left in there or hundreds of copies of the manifest.

Reply:
So does this proc if *any* thing has writing in it or if ALL of them do? I don't see why you'd check the latter and not the former, and if it's the former the writing is bad.

Reply:
I see.

```suggestion
					balloon_alert(user, "everything is written on!")
```